tests/TestCase/Database/QueryTest.php

```diff
@@ -4058,6 +4058,42 @@ public function testDirectIsNull()
         $this->assertEquals(['name' => 'larry'], $results->fetch('assoc'));
     }
 
+    /**
+     * Tests that using the wrong NULL operator will throw meaningful exception instead of
+     * cloaking as always-empty result set.
+     *
+     * @return void
+     */
+    public function testIsNullInvalid()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+
+        $this->loadFixtures('Authors');
```

Comment:
could you add/check exception message too ?
now tracking source of this exception is bit hard

Reply:
Done, now we can probably merge.